src/main/scala/com/fulcrumgenomics/bam/api/SamRecord.scala

```diff
@@ -227,6 +229,14 @@ trait SamRecord {
       refIndex == mateRefIndex &&
       SamPairUtil.getPairOrientation(this) == PairOrientation.FR
   }
+
+  /** Exposes clone() publicly for use. */
+  override def clone(): SamRecord = {
+    val r = super.clone().asInstanceOf[SamRecord]
+    r.bases = this.bases.clone()
+    r.quals = this.quals.clone()
```

Comment:
does `SAMRecord.clone()` not do this?  Yikes.

Reply:
It doesn't.  Which is pretty janky.  I think they were trying to save time/memory when just modifying some other things, but it's nasty.